packages/cli/src/commands/seed.ts

```diff
@@ -0,0 +1,504 @@
+  ManagedBudgetRoles,
+  PrimitiveType,
+  SignatureType,
+  type SignerValidatorPayload,
+  type SimpleAllowListPayload,
+  type SimpleDenyListPayload,
+  StrategyType,
+} from '@boostxyz/sdk';
+import { allowListFromAddress } from '@boostxyz/sdk';
+import { MockERC20 } from '@boostxyz/test/MockERC20';
+import { accounts } from '@boostxyz/test/accounts';
+import {
+  type BudgetFixtures,
+  type DeployableTestOptions,
+  type Fixtures,
+  defaultOptions,
+  deployFixtures,
+  freshManagedBudget,
+  fundErc20,
+} from '@boostxyz/test/helpers';
+import { SolidityBytes } from 'abitype/zod';
+import {
+  type Address,
+  type Hex,
+  isAddress,
+  isHex,
+  pad,
+  parseEther,
+  size,
+  toEventSelector,
+  toFunctionSelector,
+  zeroAddress,
+} from 'viem';
+import { type ZodType, type ZodTypeDef, z } from 'zod';
+import { type Command, type Options, getDeployableOptions } from '../utils';
+export type SeedResult = {
+  erc20?: Address;
+  boostIds?: string[];
+};
+
+const DEFAULT_MNEMONIC =
+  'test test test test test test test test test test test junk';
+
+export const seed: Command<SeedResult | BoostConfig> = async function seed(
+  positionals,
+  options: Options,
+) {
+  const privateKey = options.privateKey,
+    mnemonic = options.mnemonic ?? DEFAULT_MNEMONIC,
+    _chain = options.chain || 'anvil';
+
+  const [{ config, account }, chain] = getDeployableOptions({
+    chain: _chain,
+    rpcUrl: options.rpcUrl,
+    privateKey,
+    mnemonic,
+  });
+
+  const chainId = chain.id;
+
+  if (positionals.at(0) === 'generate') {
+    return makeSeed({ account: account?.address, chainId });
+  }
+
+  if (positionals.at(0) === 'erc20') {
+    let erc20 = new MockERC20(defaultOptions, {});
+    await erc20.deploy();
+    return {
+      erc20: erc20.assertValidAddress(),
+    };
+  }
+
+  const registryAddress = BOOST_REGISTRY_ADDRESSES[chainId];
+  if (!registryAddress) {
+    throw new Error(
+      `Unable to select a deployed BoostRegistry with chain ID ${chainId}`,
+    );
+  }
+
+  const registry = new BoostRegistry({
+    config,
+    account,
+    address: registryAddress,
+  });
+
+  const coreAddress = BOOST_CORE_ADDRESSES[chainId];
+  if (!coreAddress) {
+    throw new Error(
+      `Unable to select a deployed BoostCore with chain ID ${chainId}`,
+    );
+  }
+
+  const core = new BoostCore({
+    config,
+    account,
+    address: coreAddress,
+  });
+
+  if (!positionals.length) throw new Error('No seed provided');
+  const templates = await Promise.all(positionals.map(getSeed));
+
+  let sharedBudget: ManagedBudget | undefined;
+  let sharedBudgetConfig: ManagedBudgetPayload | undefined;
+  const boostIds: string[] = [];
+  for (const template of templates) {
+    let budget: ManagedBudget;
+    if (typeof template.budget === 'string' && isAddress(template.budget))
+      budget = core.ManagedBudget(template.budget);
+    // TODO: create budget from Core
+    else if (sharedBudgetConfig === template.budget && sharedBudget) {
```

Comment:
I'm not sure `sharedBudgetConfig === template.budget` will ever be true, the template.budget will be a different object between each template

Reply:
if a budget config is the same across templates, use the same one

Reply:
i suppose I should do deep equality